### PR TITLE
chore: Collect remaining env variables in envs.rs modules

### PR DESCRIPTION
**Pull request by KoalaSat**
Fixes https://github.com/fedimint/fedimint/issues/4191

It extracts all ENV variable names in the code to `envs.rs` files

### CODE REVIEW

File: crypto/aead/src/envs.rs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Env variable to make the creation Argon2 easier for testing
+pub const FM_TEST_FAST_WEAK_CRYPTO: &str = "FM_TEST_FAST_WEAK_CRYPTO";
```
Comment (dpc) [inline crypto/aead/src/envs.rs:2]: I think we've been adding `_ENV` to the existing env var name constants.

**Review by dpc — DISMISSED** [review]
While it's a bit of extra overhead, you'll have easier time getting reviews if you do refactorings in parts (e.g. per-crate or a group of crates). Constant names could be a bit more consistent to what we've did in other envs, but ultimately it doesn't matter all that much.

**Comment by KoalaSat** [comment]
> While it's a bit of extra overhead, you'll have easier time getting reviews if you do refactorings in parts (e.g. per-crate or a group of crates). Constant names could be a bit more consistent to what we've did in other envs, but ultimately it doesn't matter all that much.

Alright thanks, then I'll just continue working on refactoring and removing duplication on this draft. After that I can just start creating smaller PRs, one per crate.

I believe, this was important to visualize real duplications anyways.

Question: where would you say it's the best place to locate shared env variables between `/devimint| and `/fedimint`?

**Comment by dpc** [comment]
> Alright thanks, then I'll just continue working on refactoring and removing duplication on this draft. After that I can just start creating smaller PRs, one per crate.

Don't worry. That's just a mildly relevant general advice. :)


> Question: where would you say it's the best place to locate shared env variables between /devimint| and /fedimint`?

That depends on the env-var and how it's used. Possibly in some cases we might just want to keep them redundant. 

> I believe, this was important to visualize real duplications anyways.

In this PR I think I'm more interested in removing magic constants all over the code, and gathering env variable in one module. I wouldn't stress too much about duplication, especially if removing it would require adding extra dependencies to be able to share. 


We might consider introducing extra crates just to be able to share env vars, but seems ... like a lot of ceremony for little gain.

**Comment by KoalaSat** [comment]
@dpc I left a lot of `// Env variable to TODO`, I'll create a different PR for it to avoid too much rebasing for this branch

**Review by dpc — APPROVED** [review]

**Review by maan2003 — APPROVED** [review]

**Comment by KoalaSat** [comment]
@maan2003 looks like the Linux build failed but it doesn't look related to my PR, any insight?

**Comment by maan2003** [comment]
restarting, maybe just a flake

File: devimint/src/envs.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
+// faucet.rs
+
+// Env variable to TODO
+pub const FM_FAUCET_BIND_ADDR_ENV: &str = "FM_FAUCET_BIND_ADDR";
+
+// Env variable to TODO
+pub const FM_BITCOIN_RPC_URL_ENV: &str = "FM_PASSWORD";
```
Comment (dpc) [inline devimint/src/envs.rs:7]: Mistake or intentional?

File: fedimintd/src/envs.rs [hunk 3]
```diff
@@ -1 +1,38 @@
+// Env variable to TODO
 pub const FM_DISABLE_META_MODULE_ENV: &str = "FM_DISABLE_META_MODULE";
+
+// Env variable to TODO
+pub const FM_EXTRA_DKG_META_VAR_ENV: &str = "FM_EXTRA_DKG_META";
```
Comment (dpc) [inline fedimintd/src/envs.rs:5]: `_VAR_ENV` is not like other ones (`_VAR`).

**Comment by dpc** [comment]
I think I've found a mistake.

File: devimint/src/envs.rs (hunk 2 above)
Comment (KoalaSat) [reply devimint/src/envs.rs]: mistake, fixed

File: fedimintd/src/envs.rs (hunk 3 above)
Comment (KoalaSat) [reply fedimintd/src/envs.rs]: changed

**Review by dpc — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

**Pull request merged**